frontend/public/components/app.jsx

```diff
@@ -303,6 +305,24 @@ const CaptureTelemetry = React.memo(function CaptureTelemetry() {
   return null;
 });
 
+const SetTheme = React.memo(function SetTheme() {
+  const [theme, setTheme, themeLoaded] = useUserSettings('console.theme');
+  const mqListener = React.useCallback(
+    (e) => {
+      setTheme(e.matches ? 'dark' : 'light');
+    },
+    [setTheme],
+  );
+  React.useEffect(() => {
+    const htmlTagElement = document.getElementsByTagName('html')[0];
+    const darkThemeMq = window.matchMedia('(prefers-color-scheme: dark)');
+    darkThemeMq.addListener(mqListener);
```

Comment:
The matchMedia listener should only be used when the theme is set to system default.